plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/DefaultGlueColumnStatisticsProvider.java

```diff
@@ -0,0 +1,266 @@
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public Map<String, HiveColumnStatistics> getPartitionColumnStatistics(Partition partition)
+    {
+        try {
+            List<List<Column>> columnChunks = Lists.partition(partition.getColumns(), GLUE_COLUMN_READ_STAT_PAGE_SIZE);
+            ImmutableList.Builder<CompletableFuture<GetColumnStatisticsForPartitionResult>> getStatsFutures = ImmutableList.builder();
+
+            columnChunks.forEach(partialColumns -> getStatsFutures.add(CompletableFuture.supplyAsync(() -> {
+                List<String> columnsNames = partialColumns.stream()
+                        .map(Column::getName)
+                        .collect(toImmutableList());
+
+                GetColumnStatisticsForPartitionRequest request = new GetColumnStatisticsForPartitionRequest()
+                        .withCatalogId(catalogId)
+                        .withDatabaseName(partition.getDatabaseName())
+                        .withTableName(partition.getTableName())
+                        .withColumnNames(columnsNames)
+                        .withPartitionValues(partition.getValues());
+
+                return glueClient.getColumnStatisticsForPartition(request);
+            }, readExecutor)));
+
+            HiveBasicStatistics tableStatistics = getHiveBasicStatistics(partition.getParameters());
+            ImmutableMap.Builder<String, HiveColumnStatistics> columnStatsMapBuilder = ImmutableMap.builder();
+            for (CompletableFuture<GetColumnStatisticsForPartitionResult> future : getStatsFutures.build()) {
+                GetColumnStatisticsForPartitionResult partitionColumnStats = getFutureValue(future, TrinoException.class);
+                for (ColumnStatistics columnStatistics : partitionColumnStats.getColumnStatisticsList()) {
+                    columnStatsMapBuilder.put(
+                            columnStatistics.getColumnName(),
+                            fromGlueColumnStatistics(columnStatistics.getStatisticsData(), tableStatistics.getRowCount()));
+                }
+            }
+            return columnStatsMapBuilder.build();
+        }
+        catch (AmazonServiceException ex) {
+            throw new TrinoException(HIVE_METASTORE_ERROR, ex);
+        }
+    }
+
+    @Override
+    public void updateTableColumnStatistics(Table table, Map<String, HiveColumnStatistics> updatedTableColumnStatistics)
+    {
+        try {
+            HiveBasicStatistics tableStats = getHiveBasicStatistics(table.getParameters());
+            List<ColumnStatistics> columnStats = toGlueColumnStatistics(table, updatedTableColumnStatistics, tableStats.getRowCount());
+            List<List<ColumnStatistics>> columnChunks = Lists.partition(columnStats, GLUE_COLUMN_WRITE_STAT_PAGE_SIZE);
+
+            List<CompletableFuture<Void>> writeChunkFutures = columnChunks.stream().map(columnChunk -> runAsync(
```

Comment:
is there a reason to do updates first and removals later? It feels to me it would be better to gather the futures on single list and call `getFutureValue(allOf` just once. 
So even in case of an error for some of the calls we have as many changes applied as possible.